### PR TITLE
Add the internalName field to TownModel and other fixes

### PR DESCRIPTION
**Pull request by ShishkinP**
Various refactors, fixes, deduplications in the externalization code.

### CODE REVIEW

File: src/externalized/mercs/MercProfile.cc [hunk 1]
```diff
@@ -139,7 +140,7 @@ std::unique_ptr<MERCPROFILESTRUCT> MercProfile::deserializeStruct(const JsonValu
 	}
 	prof->ubCivilianGroup = Internals::getCivilianGroupEnumFromString(r.getOptionalString("civilianGroup"));
 	if (r.getOptionalBool("isGoodGuy")) {
-		prof->ubMiscFlags |= PROFILE_MISC_FLAG3_GOODGUY;
+		prof->ubMiscFlags3 |= PROFILE_MISC_FLAG3_GOODGUY;
```
Comment (lynxlynxlynx) [inline src/externalized/mercs/MercProfile.cc:143]: What's this about? Looks unrelated to the rest of the changes.

File: rust/stracciatella/src/schemas/yaml/strategic-mines.schema.yaml [hunk 2]
```diff
@@ -14,10 +14,10 @@ items:
       title: Entrance Sector
```
Comment (lynxlynxlynx) [inline rust/stracciatella/src/schemas/yaml/strategic-mines.schema.yaml:14]: can you split this one into the previous commits, so they'll be buildable as well?

File: src/externalized/mercs/MercProfile.cc (hunk 1 above)
Comment (ShishkinP) [reply src/externalized/mercs/MercProfile.cc:143]: Quick typo fix. Buns currently doesn't have weapons displayed on the AIM site because of this.
